future/name-resolution/recursive-dns/README.md

```diff
@@ -625,7 +625,135 @@ NOTE: We hacked this together a little... We used glue records in the zonefile w
 
 ## Watch the recursive DNS lookup using `tcpdump`
 
-<!-- TODO -->
+So far, we've been building out and testing this system while relying on a piece of achitecture that we haven't addressed at all yet... The recursive resolver. The role of the resolver in the DNS process is keep asking each machine it's pointed to the DNS query it's trying to resolve until it gets an answer. 
+
+Now that we've got everything setup and working, we can hopon a resolver and watch all of the requests that are necessary to make this recursive lookup work. Let's `hopon resolver-c`. 
+
+First though, the requests we made to prove that `www.awesomecat.com` was working have cached the responses to the various DNS queries. We need to clear that cache. The easiest way to do this is to restart the resolver software. The software we used for our resolver on this toy internet is called `unbound`. We'll use the same process we did to restart `knot`, namely, find the process ID with `ps aux` and then run `kill -HUP <process_id>`.
+
+Once that's done, we can run our `tcpdump`. `resolver-c` only has one interface, so we can run a simple `tcpdump -n` to see all the packets running through that interface. Open a second window, `hopon client-c1` in that window, and run `dig www.awesomecat.com`. You should see A LOT of output in your `tcpdump`. Let's take it line by line:
+
+```bash
+21:11:07.561149 ARP, Request who-has 1.2.0.100 tell 1.2.0.3, length 28
+21:11:07.561667 ARP, Reply 1.2.0.100 is-at 02:42:01:02:00:64, length 28
+```
+
+A standard ARP request. If you look at your network map, you'll see that `resolver-c` has IP Address `1.2.0.100`. So `1.2.0.3` (`router-c3`) is asking which machine on this network is `1.2.0.100`. The next line is `resolver-c` replying to `router-c3`'s ARP request with it's own MAC address. For more on this, checkout the [IP and MAC addresses appendix](../../../appendix/ip-and-mac-addresses.md)
+
+```bash
+21:11:07.561754 IP 1.1.0.200.48600 > 1.2.0.100.53: 48085+ [1au] A? www.awesomecat.com. (59)
+```
+
+<!-- TODO: maybe define A record beyond just the paranthetical? Or do an aside on record types?  -->
+`1.1.0.200` (`client-c1`) sends a request to `1.2.0.100` (`resolver-c`) port `53` requesting the `A` records (the IPv4 records) for `www.awesomecat.com`.
+
+```bash
+21:11:07.562914 IP 1.2.0.100.47301 > 101.0.1.100.53: 35603% [1au] NS? . (28)
+21:11:07.564677 IP 101.0.1.100.53 > 1.2.0.100.47301: 35603*- 2/0/3 NS rootdns-i.isc.org., NS rootdns-n.netnod.org. (123)
+```
+
+<!-- TODO: define NS/SOA records -->
+`1.2.0.100` (`resolver-c`) sends a request (`>`) to `101.0.1.100` (`rootdns-n`) for the `NS` records for `.`, the root of all DNS. Then, `101.0.1.100` (`rootdns-n`) sends a reply back to `1.2.0.100` (`resolver-c`) providing the `NS` records for the root DNS servers.
```

Comment:
> for the `NS` records for `.`, the root of all DNS.

and 

> providing the `NS` records for the root DNS servers.

I'm confused. It's my understanding that `resolver-c` already knew where the root DNS servers were (from its `root.hints` file). So what is happening here is that it's now got information on what all the TLD servers are, correct? That's why it begins sending requests to all the TLD servers next.